builtin/credential/ldap/backend.go

```diff
@@ -183,10 +191,11 @@ func (b *backend) Login(req *logical.Request, username string, password string)
 	}
 
 	if len(policies) == 0 {
-		return nil, logical.ErrorResponse("user is not member of any authorized group"), nil
+		resp.Data["error"] = "user is not a member of any authorized group"
```

Comment:
The sting "no group DN..." will be replicated in both `warnings` field and in resp.Data["error"] field. If you left it knowingly, then its fine. I only wanted to bring your attention on this.

Initially we were error-ing. Now that we are not, policies and error are nil here.
But, we are doing `if len(policies) == 0 { return resp, err }` in the calling function.
I feel it should be `if policies == nil { return resp, nil }`.